src/ir/mod.rs

```diff
@@ -552,16 +620,25 @@ pub enum LoadKind {
     // ambiently available, we probably want to trim this down to just "value"
     // and then maybe some sign extensions. We'd then use the type of the node
     // to figure out what kind of store it actually is.
-    I32,
-    I64,
+    I32 { atomic: bool },
+    I64 { atomic: bool },
     F32,
     F64,
     V128,
-    I32_8 { sign_extend: bool },
-    I32_16 { sign_extend: bool },
-    I64_8 { sign_extend: bool },
-    I64_16 { sign_extend: bool },
-    I64_32 { sign_extend: bool },
+    I32_8 { kind: ExtendedLoad },
+    I32_16 { kind: ExtendedLoad },
+    I64_8 { kind: ExtendedLoad },
+    I64_16 { kind: ExtendedLoad },
+    I64_32 { kind: ExtendedLoad },
+}
+
+/// The kinds of extended loads which can happen
+#[derive(Debug, Copy, Clone)]
+#[allow(missing_docs)]
+pub enum ExtendedLoad {
+    SignExtend,
+    ZeroExtend,
+    ZeroExtendAtomic,
```

Comment:
Huh, so there aren't any sign extending atomics?

Reply:
I was also surprised when writing, but I couldn't find them [in the spec](https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md). What I did find, however, was [this section](https://github.com/WebAssembly/threads/blob/master/proposals/threads/Overview.md#new-sign-extending-operators) "New Sign-extending Operators". I think that the CG must have decided that unsigned ops are the way to go from now on and sign extension should go through https://github.com/WebAssembly/sign-extension-ops/, dedicated instructions to sign extend values